atomic_reactor/config.py

```diff
@@ -407,22 +408,28 @@ def registries_organization(self):
         return self._get_value(ReactorConfigKeys.REGISTRIES_ORGANIZATION_KEY, fallback=None)
 
     @property
-    def registries(self):
-        all_registries = self._get_value(ReactorConfigKeys.REGISTRIES_KEY)
+    def registry(self):
+        if self._registry:
+            return self._registry
+        all_registries = list(self._get_value(ReactorConfigKeys.REGISTRIES_KEY))
 
-        registries_cm = {}
-        for registry in all_registries:
-            reguri = RegistryURI(registry.get('url'))
-            regdict = {}
-            regdict['version'] = reguri.version
-            if registry.get('auth'):
-                regdict['secret'] = registry['auth']['cfg_path']
-            regdict['insecure'] = registry.get('insecure', False)
-            regdict['expected_media_types'] = registry.get('expected_media_types', [])
+        if not all_registries:
+            return self._registry
+
+        registry = all_registries[0]
+
+        reguri = RegistryURI(registry.get('url'))
+        regdict = {'uri': reguri.docker_uri, 'version': reguri.version}
+        if registry.get('auth'):
+            regdict['secret'] = registry['auth']['cfg_path']
+        regdict['insecure'] = registry.get('insecure', False)
+        regdict['expected_media_types'] = registry.get('expected_media_types', [])
+        regdict['digests'] = {}
+        regdict['config'] = None
```

Comment:
Why do we need digests and config? The original property didn't have them

Reply:
it was part `DockerRegistry` class. which was used as a member of `PushConf` which would store the registry, their digests and config.

Reply:
In that case we should not remove `DockerRegistry` and `PushConf`. I think the idea behind this change was that nothing in those classes would be needed anymore, including digests and config.

@rcerven is that not the case?

Reply:
you shouldn't store anymore digests & config, just query it directly via registry api when needed, it is used just in 2 koji plugins

Reply:
this registry should be really just registry itself and not for storing data for images ....

Reply:
changed it to query the registry directly instead of storing it.